src/Features/Core/Portable/FindUsages/DefinitionItemFactory.cs

```diff
@@ -151,75 +120,129 @@ internal static class DefinitionItemExtensions
             {
                 return [];
             }
 
-            // If it's a namespace, don't create any normal location.  Namespaces
-            // come from many different sources, but we'll only show a single 
-            // root definition node for it.  That node won't be navigable.
-            using var sourceLocations = TemporaryArray<DocumentSpan>.Empty;
+            var assembly = definition as IAssemblySymbol ?? definition.ContainingAssembly;
+            if (assembly != null)
+            {
+                // symbol is defined within a single metadata assembly:
+                return [GetMetadataLocation(assembly, solution, out originatingProjectId)];
+            }
 
-            foreach (var location in locations)
+            if (definition is INamespaceSymbol namespaceSymbol)
             {
-                if (location.IsInMetadata)
+                using var metadataLocations = TemporaryArray<AssemblyLocation>.Empty;
+
+                // Global namespace has a metadata location for each referenced assembly.
+                // It is not useful to display these locations.
+                if (namespaceSymbol.IsGlobalNamespace)
                 {
-                    return default;
+                    return [];
                 }
 
-                if (location.IsInSource)
+                // only shared namespace symbols don't have containing assembly:
+                Contract.ThrowIfTrue(namespaceSymbol.ConstituentNamespaces.IsEmpty);
+
+                foreach (var constituentNamespace in namespaceSymbol.ConstituentNamespaces)
                 {
-                    if (!location.IsVisibleSourceLocation() &&
-                        !includeHiddenLocations)
+                    // skip source namespace definitions:
+                    if (!constituentNamespace.Locations.Any(static location => location.MetadataModule != null))
                     {
                         continue;
                     }
 
-                    var document = solution.GetDocument(location.SourceTree);
-                    if (document != null)
-                    {
-                        sourceLocations.Add(new DocumentSpan(document, location.SourceSpan));
-                    }
+                    var containingAssembly = constituentNamespace.ContainingAssembly;
+                    Contract.ThrowIfNull(containingAssembly);
+
+                    var info = solution.CompilationState.GetOriginatingProjectInfo(containingAssembly);
+                    Contract.ThrowIfNull(info);
+                    Contract.ThrowIfNull(info.ReferencedThrough);
+                    Debug.Assert(originatingProjectId == null || originatingProjectId == info.ProjectId);
+
+                    originatingProjectId = info.ProjectId;
+                    metadataLocations.Add(new AssemblyLocation(containingAssembly.Identity.Name, containingAssembly.Identity.Version, info.ReferencedThrough.Value.FilePath));
```

Comment:
i feel like it would be good to doc a general intuition on what's going o nhere.  basically what data was flowing in, and what we're trying to show the user about it.